packages/playwright-core/src/server/electron/electron.ts

```diff
@@ -132,11 +132,13 @@ export class ElectronApplication extends SdkObject {
   }
 
   async close() {
-    const progressController = new ProgressController(serverSideCallMetadata(), this);
-    const closed = progressController.run(progress => helper.waitForEvent(progress, this, ElectronApplication.Events.Close).promise);
-    await this._browserContext.close({ reason: 'Application exited' });
-    this._nodeConnection.close();
-    await closed;
+    if (!this._startedClosing) {
+      this._startedClosing = true;
+      // This will call BrowserContext.setCustomCloseHandler.
+      await this._browserContext.close({ reason: 'Application exited' });
+      this._nodeConnection.close();
+    }
+    await this._gracefullyClose();
```

Comment:
Why do we call this? Just in case that `app.quit()` did not work? Shall we just wait for the process exit instead? Shall we move this to the custom close handler?

Reply:
Yes, I don't trust `app.quit()`, since we catch it anyways, so gracefullyClose felt better. Moved it into the close handler.